src/modules/ekf2/EKF/aid_sources/range_finder/range_finder_consistency_check.cpp

```diff
@@ -36,56 +36,119 @@
  */
 
 #include <aid_sources/range_finder/range_finder_consistency_check.hpp>
+#include "ekf_derivation/generated/range_validation_filter.h"
 
-void RangeFinderConsistencyCheck::update(float dist_bottom, float dist_bottom_var, float vz, float vz_var,
-		bool horizontal_motion, uint64_t time_us)
+using namespace matrix;
+
+void RangeFinderConsistencyCheck::init(const float &z, const float &z_var, const float &dist_bottom,
+				       const float &dist_bottom_var)
 {
-	if (horizontal_motion) {
-		_time_last_horizontal_motion = time_us;
-	}
+	float p[4] = {z_var, 0.f, 0.f, z_var + dist_bottom_var};
+	_P = Matrix<float, RangeFilter::size, RangeFilter::size>(p);
+	sym::RangeValidationFilter(&_H);
```

Comment:
if you set the output to be None in symforce it will create a function that returns H instead of having to pass it via a pointer (https://github.com/PX4/PX4-Autopilot/blob/defccfa99b01b74f106b84f7b778d46fc992a469/src/modules/ekf2/EKF/python/ekf_derivation/derivation.py#L738)